app/src/main/java/org/mozilla/fenix/search/SearchDialogController.kt

```diff
@@ -74,12 +74,12 @@ class SearchDialogController(
                 navController.navigateSafe(R.id.searchDialogFragment, directions)
             }
             "moz://a" -> openSearchOrUrl(SupportUtils.getMozillaPageUrl(SupportUtils.MozillaPage.MANIFESTO))
-            else -> if (url.isNotBlank()) {
-                openSearchOrUrl(url)
-            } else {
-                dismissDialog()
-            }
+            else ->
+                if (url.isNotBlank()) {
+                    openSearchOrUrl(url)
```

Comment:
All of these other methods call `clearToolbarFocus` which I assume then dismiss the dialog somehow? Do we know where that happens and why it doesn't work for certain devices? Or is it a race that can happen on any device?

Reply:
I am not completely sure calling `clearToolbarFocus` would dismiss the dialog on all platforms and devices. It is not mentioned in the documentation for Dialog Fragment as a way to dismiss the dialog. I could not reproduce the bug on multiple devices, but this speculative fix should work in those instances where the dialog is still shown. And for the others, it should not change anything as it checks if the dialog is not already dismissed.